polyfill/test/PlainDate/prototype/subtract/balance-smaller-units.js

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.dateadd
+description: Durations with units smaller than days are balanced before adding, in the calendar
+includes: [temporalHelpers.js]
+features: [Temporal]
+---*/
+
+const actual = [];
+
+class DateAddCalendar extends Temporal.Calendar {
+  constructor() {
+    super("iso8601");
+  }
+
+  dateAdd(date, duration, options) {
+    actual.push(duration);
+    return super.dateAdd(date, duration, options);
+  }
+}
+
+const calendar = new DateAddCalendar();
+const date = new Temporal.PlainDate(2000, 5, 2, calendar);
+const duration = new Temporal.Duration(0, 0, 0, 1, 24, 1440, 86400, 86400_000, 86400_000_000, 86400_000_000_000);
+
+const result = date.subtract(duration);
+TemporalHelpers.assertPlainDate(result, 2000, 4, "M04", 25, "units smaller than days are balanced");
+
+assert.sameValue(actual.length, 1, "calendar.dateAdd called exactly once");
+TemporalHelpers.assertDuration(actual[0], 0, 0, 0, -1, -24, -1440, -86400, -86400_000, -86400_000_000, -86400_000_000_000, "the duration is negated but not balanced before passing to the calendar");
```

Comment:
Do we want to check that `actual[0] !== duration`? Seems like a fairly improbable mess-up, but still.